pkg/ddl/backfilling.go

```diff
@@ -845,12 +846,20 @@ func (dc *ddlCtx) writePhysicalTableRecord(
 	t table.PhysicalTable,
 	bfWorkerType backfillerType,
 	reorgInfo *reorgInfo,
-) error {
+) (err error) {
 	startKey, endKey := reorgInfo.StartKey, reorgInfo.EndKey
 
 	if err := dc.isReorgRunnable(reorgInfo.Job.ID, false); err != nil {
 		return errors.Trace(err)
 	}
+	defer func() {
+		if err != nil && ctx.Err() != nil {
+			// Prevent "not owner" error being overwritten by context.Canceled.
```

Comment:
maybe use `WithCancelCause` to create the context, and use `cancel(dbterror.ErrNotOwner)` to cancel